src/dash-table/syntax-tree/SingleColumnSyntaxTree.ts

```diff
@@ -23,29 +36,38 @@ function isUnary(lexemes: ILexemeResult[]) {
         lexemes[0].lexeme.type === LexemeType.UnaryOperator;
 }
 
-export function modifyLex(key: ColumnId, res: ILexerResult) {
+export function modifyLex(config: SingleColumnConfig, res: ILexerResult) {
     if (!res.valid) {
         return res;
     }
 
     if (isBinary(res.lexemes) || isUnary(res.lexemes)) {
         res.lexemes = [
-            { lexeme: boundLexeme(operand), value: `{${key}}` },
+            { lexeme: boundLexeme(operand), value: `{${config.id}}` },
             ...res.lexemes
         ];
     } else if (isExpression(res.lexemes)) {
         res.lexemes = [
-            { lexeme: boundLexeme(operand), value: `{${key}}` },
-            { lexeme: boundLexeme(equal), value: 'eq' },
+            { lexeme: boundLexeme(operand), value: `{${config.id}}` },
+            {
+                lexeme: boundLexeme(equal),
+                value: getDefaultRelationalOperator(config.type)
+            },
             ...res.lexemes
         ];
     }
 
     return res;
 }
 
+export type SingleColumnConfig = RequiredPluck<IVisibleColumn, 'id'> & OptionalPluck<IVisibleColumn, 'type'>;
```

Comment:
`Partial<IVisibleColumn>` was problematic as `id` had to be redefined on top, causing potential misalignment issues + exposed the entire column interface for no good reason. Plucking and optionally plucking props from the type instead to create a new tailor-made partial type.